### DISCUSSION TITLE
Extract or save the code of the component partially

### DISCUSSION THREAD

**Discussion by HonTung** [GrapesJS/grapesjs#5177] [Ideas]
Hello Grapesjs Team!

I feel so great when found GrapesJS, and now I am still in a exploring stage.

Just wondering, is there any hook or function that I could extract and save the code of the component or export from Layer Manager? So I can make them storing into block manager library/database and reusable when editing new template.

![component-save-function-sample-visual](https://github.com/GrapesJS/grapesjs/assets/538491/ebf17d8a-74c1-483d-9b70-e692866259ae)

Thanks 😊

**Comment by ronaldohoch** [comment]
Hello, you can use this to save the json of selected component:
`JSON.stringify(editor.getSelected())`

But i don't know how to get the style from selected component and their children..

**Reply by ronaldohoch** [reply]
Maybe with this you can get the styles: https://grapesjs.com/docs/api/css_composer.html#getrules

**Comment by HonTung** [comment]
`Thanks` @ronaldohoch for the hint. 

I have created simple grapesjs command script that works

```
editor.Commands.add('save-component',{
       run: function(editor){
       const component = editor.getSelected();
       const componentMJML = `${editor.getHtml({ component })}`;
       console.log( componentMJML );
   }
});
```

Hope one day I could build one plugin function for saving the component.

**Reply by ronaldohoch** [reply]
niice!
Didn't know this works too, thank you!
